mm2src/mm2_main/src/mm2.rs

```diff
@@ -160,10 +160,30 @@ pub async fn lp_main(
         .with_datetime(datetime.clone())
         .into_mm_arc();
     ctx_cb(try_s!(ctx.ffi_handle()));
+
+    #[cfg(not(target_arch = "wasm32"))]
+    spawn_ctrl_c_handler(ctx.clone());
+
     try_s!(lp_init(ctx, version, datetime).await);
     Ok(())
 }
 
+/// Handles CTRL-C signals and shutdowns the KDF runtime gracefully.
+///
+/// It's important to spawn this task as soon as `Ctx` is in the correct state.
+#[cfg(not(target_arch = "wasm32"))]
+fn spawn_ctrl_c_handler(ctx: mm2_core::mm_ctx::MmArc) {
+    common::executor::spawn(async move {
+        tokio::signal::ctrl_c()
+            .await
+            .expect("Couldn't listen for the CTRL-C signal.");
+
+        log::info!("Wrapping things up and shutting down...");
+
+        ctx.stop().await.expect("Couldn't stop the KDF runtime.");
```

Comment:
What about handling the below first
https://github.com/KomodoPlatform/komodo-defi-framework/blob/7723b50fcb533ebe12b621eb862b8a958113696b/mm2src/mm2_main/src/lp_ordermatch/lp_bot.rs#L217 Similar to how it's done in stop rpc from here
https://github.com/KomodoPlatform/komodo-defi-framework/blob/7cc515749347afb821dc9937a48fc0b30d68ca37/mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs#L245